Pass on readme

VictorSanh opened the pull request:
Re-organize REAME, add documentation and update contribution guidelines.

Also `get_fixed_answer_choices_list` seems redundant (we are never using it) so I removed it.

Once we have a "overview figure" of promptsource, I will add it to the readme too

srush commented on README.md:
```diff
@@ -1,112 +1,113 @@
 
-Promptsource uses a simple templating language to programatically map an example of a dataset into a text input and a text target.
+Prompts are functions that map an example from a dataset to a natural language input and target output PromptSource contains a growing collection of prompts (which we call **P3**: **P**ublic **P**ool of **P**rompts). As of January 20, 2022, there are ~2'000 English prompts for 170+ English datasets in [P3](https://huggingface.co/datasets/bigscience/P3).
 
```
I think some popular language here about how prompting is really important because of GPT-3 and then T0 / FLAN, and how it is a new type of NLP tool with lots of interest etc. Would be really nice.

srush commented on README.md:
````diff
@@ -1,112 +1,113 @@
-If you want to launch a local version (in particular to write propmts, from the root directory of the repo, launch the editor with:
-```
-streamlit run promptsource/app.py
+## How to use prompts
````
This is really nice. I like that your started with this.

srush commented on README.md:
```diff
@@ -1,112 +1,113 @@
-3. Install requirements with `pip install -r requirements.txt` in a Python 3.7 environment
-4. Run `pip install -e .` to install the `promptsource` module
+1. Navigate to the root directory of the repo
+1. Run `pip install -e .` to install the `promptsource` module
```
Can we put it in PIP? Do you think most people will contribute new prompts?

VictorSanh replied:
yes, i will put in pip as soon as we have pinned `v0.2`.

srush commented on README.md:
```diff
@@ -1,112 +1,113 @@
-**You will find the official repository to reproduce the results of T Zero here: https://github.com/bigscience-workshop/t-zero.**
+PromptSource provides the tools to create, and share natural language prompts (see [How to use prompts](#how-to-use-prompts), and then use the thousands of existing and newly created prompts through a simple API (see How to use prompts).
+Prompts are saved in standalone structured files and are written in a simple templating language called Jinja.
 
```
Needs an example of a prompt here. Also we should include figure 1 from the paper, it is awesome way to explain what is going on.

srush commented on README.md:
```diff
@@ -1,112 +1,113 @@
 # PromptSource
```
Logo of some sort...

srush commented on README.md:
```diff
@@ -1,112 +1,113 @@
-
-**You will find the official repository to reproduce the results of T Zero here: https://github.com/bigscience-workshop/t-zero.**
+PromptSource provides the tools to create, and share natural language prompts (see [How to use prompts](#how-to-use-prompts), and then use the thousands of existing and newly created prompts through a simple API (see How to use prompts).
+Prompts are saved in standalone structured files and are written in a simple templating language called Jinja.
```
Hosted version? Is it too slow?

VictorSanh commented:
Thanks @srush !
addressed all comments except for logo.
Once we have the arxiv version uploaded, I'll updated the citation info.
Will merge then.
And then will pin v0.2 and put on pip.

srush commented on README.md:
```diff
@@ -1,112 +1,132 @@
+**PromptSource is a toolkit for creating, sharing and using natural language prompts.**
 
-Promptsource uses a simple templating language to programatically map an example of a dataset into a text input and a text target.
+Recent work has shown that large language models exhibit the ability to perform reasonable zeor-shot generalization to new tasks. For instance, [GPT-3](https://arxiv.org/abs/2005.14165) demonstrated that large language models have strong zero- and few-shot abilities. [FLAN](https://arxiv.org/abs/2109.01652) and [T0](https://arxiv.org/abs/2110.08207) then demonstrated that pre-trained language mod ...
```
"zeor" typo
